utils/src/range/mod.rs

```diff
@@ -469,13 +469,30 @@ impl<T: Copy + Ord> BitXor<RangeSet<T>> for RangeSet<T> {
     }
 }
 
+impl<T: Copy + Ord> BitXor<&RangeSet<T>> for RangeSet<T> {
+    type Output = RangeSet<T>;
+
+    fn bitxor(mut self, rhs: &RangeSet<T>) -> Self::Output {
```

Comment:
Don't you need to also return that part of `rhs` which has no intersection with `self`?
https://en.wikipedia.org/wiki/Exclusive_or